Foundation/src/Environment_WIN32U.cpp

```diff
@@ -209,31 +209,65 @@ void EnvironmentImpl::nodeIdImpl(NodeId& id)
 	// Make an initial call to GetAdaptersInfo to get
 	// the necessary size into len
 	DWORD rc = GetAdaptersInfo(pAdapterInfo, &len);
-	if (rc == ERROR_BUFFER_OVERFLOW)
-	{
-		delete [] reinterpret_cast<char*>(pAdapterInfo);
-		pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(new char[len]);
-	}
-	else if (rc != ERROR_SUCCESS)
-	{
-		delete[] reinterpret_cast<char*>(pAdapterInfo);
-		throw SystemException("cannot get network adapter list");
-	}
-	if (GetAdaptersInfo(pAdapterInfo, &len) == NO_ERROR)
+
+	try
 	{
-		pAdapter = pAdapterInfo;
+		if (rc == ERROR_BUFFER_OVERFLOW)
+		{
+			delete[] reinterpret_cast<char*>(pAdapterInfo);
+			pAdapterInfo = reinterpret_cast<IP_ADAPTER_INFO*>(new char[len]);
```

Comment:
If new throws now we get double free. These things are always hard to check. Why won't we just use smart pointer for pAdapterInfo so we do not need to think how these kind of things go.

Reply:
> If new throws we get double free

I'm not seeing a way that would happen. The catch is only for Poco `SystemException`s and if another exception is thrown, the final `delete[]` won't execute.

Reply:
Yeah mistake from my part that I did not checked expection type close enogh.